### PR TITLE
Do not submit emails validation

### PR DESCRIPTION
**Pull request by kuzmany**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | x
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
A coporation marketer requires do not submit form for certain emails. This PR added new Do Not Submit emails list to configuration and optional validation  for this list
This feature is offered by [Salesforce Pardot](https://www.pardot.com/blog/preventing-competitors-from-accessing-content-on-form-submission/).

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Go to Configuration > Form Settings > Do not submit these emails
- add email example@domain.tld (should blocked this email)
- add domain with * (*@gmail.com) - should blocked all gmail emails
3. Create form with email enable Do Not Submit  validation. Also try customization of validation message
4. Then try submit with emails
5. Form should return error message

Screenshot

image

image

### CODE REVIEW

File: app/bundles/FormBundle/Translations/en_US/validators.ini [hunk 1]
```diff
@@ -10,6 +10,7 @@ mautic.form.submission.captcha.invalid="The answer to %label% is incorrect. Plea
 mautic.form.submission.email.invalid="The email is invalid."
 mautic.form.submission.url.invalid="The URL is invalid."
 mautic.form.submission.phone.invalid="Please use the following international phone number format [+][country code][subscriber number] for this field (eg: ‪+14028650000)."
+mautic.form.submission.email.donotsubmit.invalid="Cannot be sent with this email"
```
Comment (npracht) [inline app/bundles/FormBundle/Translations/en_US/validators.ini]: Is that variable (in form email field)? That should be a customizable text on the form if the feature is enabled on the form itself. The feature should be togglable form by form (for instance, no issue for newsletter subscription, but not acceptable to get pricelist).

Comment (kuzmany) [reply app/bundles/FormBundle/Translations/en_US/validators.ini]: It's default, customizable text

**Comment by npracht** [comment]
@escopecz funny thing, I've just received the newsletter of Mautic Inc. and your team proposes to follow this process: https://mautic.com/help/verifying-email-domains-on-form-submissions/?utm_source=july19newsletter&utm_medium=email
which is the same feature as the one of this PR. Can you kindly ask you marketing team to test the PR when finished ?

**Comment by escopecz** [comment]
Interesting. I believe that the goal is the same but that article describes how to implement a browser side validation without a need to modify Mautic. On the other hand this PR implements it on the server side.

**Comment by MajesteitBart** [comment]
It would be so great if the list of domains could be filled dynamically. I've now imported this list, which contains over 50k domains of disposable domains. Works perfectly! 
[](https://github.com/ivolo/disposable-email-domains/blob/master/index.json)

**Pull request merged**